scipy/optimize/minpack.py

```diff
@@ -230,15 +232,11 @@ def _root_hybr(func, x0, args=(), jac=None,
                   "ten iterations.", ValueError],
               'unknown': ["An error occurred.", TypeError]}
 
-    if status != 1 and not full_output:
-        if status in [2, 3, 4, 5]:
-            msg = errors[status][0]
-            warnings.warn(msg, RuntimeWarning)
-        else:
-            try:
-                raise errors[status][1](errors[status][0])
-            except KeyError:
-                raise errors['unknown'][1](errors['unknown'][0])
+    if status not in [1, 2, 3, 4, 5]:
+        try:
+            raise errors[status][1](errors[status][0])
+        except KeyError:
+            raise errors['unknown'][1](errors['unknown'][0])
 
```

Comment:
As is, an exception would still be raised from a `root()` call with no particular option. I think this should not occur. For instance, the `root` interface with `lm` method avoids this, and that with `hybr` should behave the same.  IIUC, this is what @pv suggested with having `_full_output=1` as default. This solution allows to keep the code where it is now. I thought it would be nicer to just remove the `full_ouput` parameter move this code in `fsolve`. I have no strong opinion on this though. So, @person142, I'd say you can follow whatever approach you wish.